src/screens/SettingsSheet/utils.ts

```diff
@@ -1,118 +1,121 @@
+import { backupsStore, CloudBackupState } from '@/state/backups/backups';
 import { RainbowWallet } from '@/model/wallet';
-import { Navigation } from '@/navigation';
-import { BackupUserData, getLocalBackupPassword } from '@/model/backup';
-import Routes from '@/navigation/routesNames';
-import WalletBackupStepTypes from '@/helpers/walletBackupStepTypes';
-
-type WalletsByKey = {
-  [key: string]: RainbowWallet;
-};
+import { IS_ANDROID, IS_IOS } from '@/env';
+import { normalizeAndroidBackupFilename } from '@/handlers/cloudBackup';
 
 type WalletBackupStatus = {
   allBackedUp: boolean;
   areBackedUp: boolean;
   canBeBackedUp: boolean;
-  backupProvider: string | undefined;
 };
 
-export const capitalizeFirstLetter = (str: string) => {
-  return str.charAt(0).toUpperCase() + str.slice(1);
+export const hasManuallyBackedUpWallet = (wallets: ReturnType<typeof useWallets>['wallets']) => {
+  if (!wallets) return false;
+  return Object.values(wallets).some(wallet => wallet.backupType === WalletBackupTypes.manual);
 };
 
-export const checkUserDataForBackupProvider = (userData?: BackupUserData): { backupProvider: string | undefined } => {
-  let backupProvider: string | undefined = undefined;
-
-  if (!userData?.wallets) return { backupProvider };
-
-  Object.values(userData.wallets).forEach(wallet => {
-    if (wallet.backedUp && wallet.type !== WalletTypes.readOnly) {
-      if (wallet.backupType === WalletBackupTypes.cloud) {
-        backupProvider = WalletBackupTypes.cloud;
-      } else if (backupProvider !== WalletBackupTypes.cloud && wallet.backupType === WalletBackupTypes.manual) {
-        backupProvider = WalletBackupTypes.manual;
-      }
-    }
-  });
-
-  return { backupProvider };
-};
-
-export const checkWalletsForBackupStatus = (wallets: WalletsByKey | null): WalletBackupStatus => {
-  if (!wallets)
+export const checkLocalWalletsForBackupStatus = (
+  wallets: ReturnType<typeof useWallets>['wallets'],
+  backups: CloudBackups
+): WalletBackupStatus => {
+  if (!wallets || isEmpty(wallets)) {
     return {
       allBackedUp: false,
       areBackedUp: false,
       canBeBackedUp: false,
-      backupProvider: undefined,
     };
+  }
+
+  // FOR ANDROID, we need to check if the current google account also has the backup file
+  if (IS_ANDROID) {
+    return Object.values(wallets).reduce<WalletBackupStatus>(
+      (acc, wallet) => {
+        const isBackupEligible = wallet.type !== WalletTypes.readOnly && wallet.type !== WalletTypes.bluetooth;
+        const hasBackupFile = backups.files.some(
+          file => normalizeAndroidBackupFilename(file.name) === normalizeAndroidBackupFilename(wallet.backupFile ?? '')
+        );
+
+        return {
+          allBackedUp: acc.allBackedUp && hasBackupFile && (wallet.backedUp || !isBackupEligible),
+          areBackedUp: acc.areBackedUp && hasBackupFile && (wallet.backedUp || !isBackupEligible),
+          canBeBackedUp: acc.canBeBackedUp || isBackupEligible,
```

Comment:
`acc.canBeBackedUp || isBackupEligible` --> should this be `acc.canBeBackedUp && isBackupEligible`

Reply:
oops good catch ty